marklogic-data-hub-central/ui/e2e/cypress/support/pages/browse.tsx

```diff
@@ -1,22 +1,28 @@
+import 'cypress-wait-until';
+
 class BrowsePage {
 
   getSelectedEntity() {
     return cy.get('.ant-select-selection-selected-value').invoke('text')
   }
 
   selectEntity(entity: string) {
-    cy.get('#entity-select').click();
-    return cy.get('[data-cy=entity-option]').each(function (item) {
+    cy.waitUntil(() => cy.get('#entity-select')).click();
+    return cy.waitUntil(() => cy.get('[data-cy=entity-option]')).each(function (item) {
```

Comment:
Going to give this a pass now, since tests seem to pass. 
But if they do fail again in Jenkins, the best approach for locators is just to return the element back and take action on them in the test. We shouldnt be doing click action in the locator and certainly avoid using callbacks which leads to brittle tests like we see today.

Ideally we would want to split this out into "selectEntityDropDown()" and "selectEntityType(entityType: string)" and return cy.findByText(entityType). Then the test will look like
browsePage.selectEntityDropDown().click();
browsePage.selectEntityType('Customer').click()

Reply:
Ideally we would like to hide the logic from the tests and move it to the helper methods on browse, model, etc pages. So we don't duplicate the same actions browsePage.selectEntityDropDown().click();
browsePage.selectEntityType('Customer').click() in every test. This will make tests shorter, readable and stable.

Reply:
I agree that we should hide logic from the tests and in this scenario we are talking about a couple of clicks. That is what a user would do when he comes to zero state of Explorer, click open the dropdown and select an entityType. I dont see a logic involved here. 
Infact the locator is introducing a logic when there is no need to. If the intent was to have less lines of test code, then IMO robust and stable tests take priority over less lines of test code. 
These tests were not stable with such use of the locator and that is why we are having to add waitUntil and waits with this PR.

Reply:
I think we need to use more waitUntil methods for the steps which requires api and page rendering. The main reason for the test to fail is that the api call or page elements rendering takes sometimes much more time while running tests in Cypress than in real browser.